### PR TITLE
hotfix/JM-5795 deferral processing outcome not displayed

### PR DESCRIPTION
**Pull request by akikrahman1**
### JIRA link (if applicable) ###

https://centralgovernmentcgi.atlassian.net/browse/JM-5795

### Change description ###
Updating API to send the reason for deferral to front end (it is an excusal reason)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/juror/api/moj/controller/response/JurorDetailsCommonResponseDto.java [hunk 1]
```diff
@@ -100,6 +101,10 @@ public class JurorDetailsCommonResponseDto {
     @Schema(name = "Deferral date", description = "Deferral date")
     private LocalDate deferralDate;
 
+    @JsonProperty("deferralCode")
```
Comment (Ben-Edwards-cgi) [inline src/main/java/uk/gov/hmcts/juror/api/moj/controller/response/JurorDetailsCommonResponseDto.java:104]: Should be snake case

File: src/main/java/uk/gov/hmcts/juror/api/moj/controller/response/JurorDetailsCommonResponseDto.java [hunk 2]
```diff
@@ -140,22 +145,19 @@ public JurorDetailsCommonResponseDto(JurorPool jurorPool,
...
         if (this.excusalCode != null) {
-            List<ResponseExcusalController.ExcusalCodeDto> excusalCodeDtoList =
-                responseExcusalService.getExcusalReasons();
-
-            Optional<ResponseExcusalController.ExcusalCodeDto> excusalCodeReasonOpt = excusalCodeDtoList.stream()
-                .filter(code -> code.getExcusalCode().equals(this.excusalCode)).findFirst();
-
-            this.excusalDescription =
-                excusalCodeReasonOpt.map(ResponseExcusalController.ExcusalCodeDto::getDescription).orElse(null);
+            this.excusalDescription =  ExcusalCodeEnum.valueOf(this.excusalCode).getDescription();
```
Comment (Ben-Edwards-cgi) [inline src/main/java/uk/gov/hmcts/juror/api/moj/controller/response/JurorDetailsCommonResponseDto.java:156]: Do we need validation here to ensure excusal code exists within the enum?

File: src/main/java/uk/gov/hmcts/juror/api/moj/controller/response/JurorDetailsCommonResponseDto.java [hunk 3]
```diff
@@ -140,22 +145,19 @@ public JurorDetailsCommonResponseDto(JurorPool jurorPool,
...
+        }
+        // set the excusal description as front end needs it to display the deferral reason
+        if (this.deferralCode != null) {
+            this.excusalDescription =  ExcusalCodeEnum.valueOf(this.deferralCode).getDescription();
```
Comment (Ben-Edwards-cgi) [inline src/main/java/uk/gov/hmcts/juror/api/moj/controller/response/JurorDetailsCommonResponseDto.java:160]: Do we need validation here to ensure excusal code exists within the enum?

**Review by Ben-Edwards-cgi — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/juror/api/moj/controller/response/JurorDetailsCommonResponseDto.java (hunk 1 above)
Comment (akikrahman1) [reply src/main/java/uk/gov/hmcts/juror/api/moj/controller/response/JurorDetailsCommonResponseDto.java:104]: updated

File: src/main/java/uk/gov/hmcts/juror/api/moj/controller/response/JurorDetailsCommonResponseDto.java (hunk 2 above)
Comment (akikrahman1) [reply src/main/java/uk/gov/hmcts/juror/api/moj/controller/response/JurorDetailsCommonResponseDto.java:156]: updated to throw internal server error if enum value is not found in DB

File: src/main/java/uk/gov/hmcts/juror/api/moj/controller/response/JurorDetailsCommonResponseDto.java (hunk 3 above)
Comment (akikrahman1) [reply src/main/java/uk/gov/hmcts/juror/api/moj/controller/response/JurorDetailsCommonResponseDto.java:160]: updated to throw internal server error if enum value is not found in DB

**Review by Ben-Edwards-cgi — APPROVED** [review]

**Pull request merged**